sled-agent/src/rack_setup/service.rs

```diff
@@ -1084,6 +1066,113 @@ impl ServiceInner {
     }
 }
 
+/// The service plan describes all the zones that we will eventually
+/// deploy on each sled.  But we cannot currently just deploy them all
+/// concurrently.  We'll do it in a few stages, each corresponding to a
+/// version of each sled's configuration.
+///
+/// - version 1: no services running
+///              (We don't have to do anything for this.  But we do
+///              reserve this version number for "no services running" so
+///              that sled agents can begin with an initial, valid
+///              OmicronZonesConfig before they've got anything running.)
+/// - version 2: internal DNS only
+/// - version 3: internal DNS + NTP servers
+/// - version 4: internal DNS + NTP servers + CockroachDB
+/// - version 5: everything
+///
+/// At each stage, we're specifying a complete configuration of what
+/// should be running on the sled -- including this version number.
+/// And Sled Agents will reject requests for versions older than the
+/// one they're currently running.  Thus, the version number is a piece
+/// of global, distributed state.
+///
+/// For now, we hardcode the requests we make to use specific version
+/// numbers.
+struct DeployStepVersion;
+
+impl DeployStepVersion {
+    const V1_NOTHING: Generation = OmicronZonesConfig::INITIAL_GENERATION;
+    const V2_DNS_ONLY: Generation = Self::V1_NOTHING.next();
+    const V3_DNS_AND_NTP: Generation = Self::V2_DNS_ONLY.next();
+    const V4_COCKROACHDB: Generation = Self::V3_DNS_AND_NTP.next();
+    const V5_EVERYTHING: Generation = Self::V4_COCKROACHDB.next();
+}
+
+fn build_initial_blueprint_from_plan(
```

Comment:
Pretty stoked to see this here, bridging the gap between RSS and Blueprints.

Do we have any issues tracking "Eventually, we'd like the ServicePlan to be generated by the reconfigurator"? Do we still think that's possible?

Reply:
I don't think we have an issue, but we should. I do think it's possible (and probably even _preferred_, especially if RSS could start a much more minimal set of services, handoff, and let Reconfigurator do the rest).